### PR TITLE
Clean up controllers for manage repository columns [SCI-4261]

### PR DESCRIPTION
**Pull request by mlorb**
Jira ticket: [SCI-4261](https://biosistemika.atlassian.net/browse/SCI-4261)

### What was done
- `destroy` endpoints were removed from specific column type controllers: `app/controllers/repository_columns/*`
- `create` and `update` endpoints were removed from `app/controllers/repository_columns_controller.rb`

### CODE REVIEW

File: app/controllers/repository_columns_controller.rb [hunk 1]
```diff
@@ -36,85 +35,10 @@ def create_html
...
-        else
-          render json: { message: @repository_column.errors.full_messages },
-                 status: :unprocessable_entity
-        end
-      end
-    end
-  end
-
-  def edit
+  def edit_html
```
Comment (okriuchykhin) [inline app/controllers/repository_columns_controller.rb]: Why _html is needed here?

Comment (mlorb) [reply app/controllers/repository_columns_controller.rb]: To be consistent with names of other endpoints here (`index_html`, `create_html`, `destroy_html`), which are having the same role: displaying the content of `manage_column_modal`.

Comment (okriuchykhin) [reply app/controllers/repository_columns_controller.rb]: it doesn't follow Rails naming

**Comment by mlorb** [comment]
@okriuchykhin check also [PR](https://gitlab.com/biosistemika/scinote/addons/Common/-/merge_requests/117)

File: app/controllers/repository_columns_controller.rb [hunk 2]
```diff
@@ -2,19 +2,13 @@ class RepositoryColumnsController < ApplicationController
...
-  before_action :load_vars,
-                except: ACTIONS
-  before_action :load_vars_nested,
-                only: ACTIONS
-  before_action :check_create_permissions, only: :create
-  before_action :check_manage_permissions,
-                except: ACTIONS
+  before_action :load_repository
+  before_action :load_column, only: %i(edit destroy_html destroy)
+  before_action :check_create_permissions, only: :new
```
Comment (okriuchykhin) [inline app/controllers/repository_columns_controller.rb:7]: Isn't permission check for 'create' needed for all column types?

File: app/controllers/repository_columns_controller.rb [hunk 3]
```diff
@@ -2,19 +2,13 @@ class RepositoryColumnsController < ApplicationController
...
+  before_action :check_manage_permissions, only: %i(edit destroy_html destroy)
```
Comment (okriuchykhin) [inline app/controllers/repository_columns_controller.rb:8]: Isn't permission check for 'update' needed for all column types?

File: app/controllers/repository_columns/text_columns_controller.rb [hunk 4]
```diff
@@ -1,11 +1,11 @@
 # frozen_string_literal: true
 
 module RepositoryColumns
-  class TextColumnsController < BaseColumnsController
+  class TextColumnsController < RepositoryColumnsController
     include InputSanitizeHelper
-    before_action :load_column, only: %i(update destroy)
+    before_action :load_column, only: :update
```
Comment (okriuchykhin) [inline app/controllers/repository_columns/text_columns_controller.rb:6]: Move it to RepositoryColumnsController

File: app/controllers/repository_columns/list_columns_controller.rb [hunk 5]
```diff
@@ -1,10 +1,10 @@
 # frozen_string_literal: true
 
 module RepositoryColumns
-  class ListColumnsController < BaseColumnsController
-    before_action :load_column, only: %i(update destroy items)
+  class ListColumnsController < RepositoryColumnsController
+    before_action :load_column, only: %i(update items)
```
Comment (okriuchykhin) [inline app/controllers/repository_columns/list_columns_controller.rb:5]: Move it to RepositoryColumnsController

File: app/controllers/repository_columns/number_columns_controller.rb [hunk 6]
```diff
@@ -1,11 +1,11 @@
 # frozen_string_literal: true
 
 module RepositoryColumns
-  class NumberColumnsController < BaseColumnsController
+  class NumberColumnsController < RepositoryColumnsController
     include InputSanitizeHelper
-    before_action :load_column, only: %i(update destroy)
+    before_action :load_column, only: :update
```
Comment (okriuchykhin) [inline app/controllers/repository_columns/number_columns_controller.rb:6]: Move it to RepositoryColumnsController

File: app/controllers/repository_columns/status_columns_controller.rb [hunk 7]
```diff
@@ -1,11 +1,11 @@
 # frozen_string_literal: true
 
 module RepositoryColumns
-  class StatusColumnsController < BaseColumnsController
+  class StatusColumnsController < RepositoryColumnsController
     include InputSanitizeHelper
-    before_action :load_column, only: %i(update destroy items)
+    before_action :load_column, only: %i(update items)
```
Comment (okriuchykhin) [inline app/controllers/repository_columns/status_columns_controller.rb:6]: Move it to RepositoryColumnsController

File: app/controllers/repository_columns/date_time_columns_controller.rb [hunk 8]
```diff
@@ -1,11 +1,11 @@
 # frozen_string_literal: true
 
 module RepositoryColumns
-  class DateTimeColumnsController < BaseColumnsController
+  class DateTimeColumnsController < RepositoryColumnsController
     include InputSanitizeHelper
-    before_action :load_column, only: %i(update destroy)
+    before_action :load_column, only: :update
```
Comment (okriuchykhin) [inline app/controllers/repository_columns/date_time_columns_controller.rb:6]: Move it to RepositoryColumnsController

File: app/controllers/repository_columns/asset_columns_controller.rb [hunk 9]
```diff
@@ -1,11 +1,11 @@
 # frozen_string_literal: true
 
 module RepositoryColumns
-  class AssetColumnsController < BaseColumnsController
+  class AssetColumnsController < RepositoryColumnsController
     include InputSanitizeHelper
-    before_action :load_column, only: %i(update destroy)
+    before_action :load_column, only: :update
```
Comment (okriuchykhin) [inline app/controllers/repository_columns/asset_columns_controller.rb:6]: Move it to RepositoryColumnsController

**Comment by mlorb** [comment]
There are changes also in [Common](https://gitlab.com/biosistemika/scinote/addons/Common/-/merge_requests/117/diffs)

File: app/controllers/repository_columns_controller.rb [hunk 10]
```diff
@@ -151,6 +82,8 @@ def destroy
     end
   end
 
+  def items; end
```
Comment (okriuchykhin) [inline app/controllers/repository_columns_controller.rb:85]: Add 'raise NotImplementedError'

**Pull request merged**